src/components/RecipeView.vue

```diff
@@ -185,82 +186,79 @@ export default {
-                }
-                if ($this.$store.state.recipe.nutrition) {
-                    if ( $this.$store.state.recipe.nutrition instanceof Array) {
+            axios.get(this.$window.baseUrl + '/api/recipes/' + this.$route.params.id)
+                .then(function (response) {
+                    let recipe = response.data
+                    // Store recipe data in vuex
+                    $this.$store.dispatch('setRecipe', { recipe: recipe })
+
+                    if ($this.$store.state.recipe.recipeIngredient) {
+                        $this.ingredients = Object.values($this.$store.state.recipe.recipeIngredient)
+                    }
+
+                    if ($this.$store.state.recipe.recipeInstructions) {
+                        $this.instructions = Object.values($this.$store.state.recipe.recipeInstructions)
+                    }
+
+                    if ($this.$store.state.recipe.keywords) {
+                        $this.keywords = String($this.$store.state.recipe.keywords).split(',')
+                    }
+
+                    if ($this.$store.state.recipe.cookTime) {
+                        let cookT = $this.$store.state.recipe.cookTime.match(/PT(\d+?)H(\d+?)M/)
+                        $this.timerCook = { hours: parseInt(cookT[1]), minutes: parseInt(cookT[2]) }
+                    }
+
+                    if ($this.$store.state.recipe.prepTime) {
+                        let prepT = $this.$store.state.recipe.prepTime.match(/PT(\d+?)H(\d+?)M/)
+                        $this.timerPrep = { hours: parseInt(prepT[1]), minutes: parseInt(prepT[2]) }
+                    }
+
+                    if ($this.$store.state.recipe.totalTime) {
+                        let totalT = $this.$store.state.recipe.totalTime.match(/PT(\d+?)H(\d+?)M/)
+                        $this.timerTotal = { hours: parseInt(totalT[1]), minutes: parseInt(totalT[2]) }
+                    }
+
+                    if ($this.$store.state.recipe.tool) {
+                        $this.tools = $this.$store.state.recipe.tool
+                    }
+
+                    if ($this.$store.state.recipe.dateCreated) {
+                        let date = $this.parseDateTime($this.$store.state.recipe.dateCreated)
+                        $this.dateCreated = (date != null ? date.format('L, LT').toString() : null)
+                    }
+
+                    if ($this.$store.state.recipe.dateModified) {
+                        let date = $this.parseDateTime($this.$store.state.recipe.dateModified)
+                        $this.dateModified = (date != null ? date.format('L, LT').toString() : null)
+                    }
+                    if ($this.$store.state.recipe.nutrition) {
+                        if ( $this.$store.state.recipe.nutrition instanceof Array) {
+                            $this.$store.state.recipe.nutrition = {}
+                        }
+                    } else {
```

Comment:
This might be well suited for a computed property from the recipe data itself.

Reply:
yes. what do you think is better: a recipe object as a computed property or a computed property for each recipe part (ingredients, dateModified, etc.)?

Reply:
I assume both are similarly useful. Personally, I assume a single property with an object might be simpler to use if we think what might (or might not) be able to extend/abstract the component.